ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/manager/EthScheduler.java

```diff
@@ -194,25 +218,46 @@ public void stop() {
     if (stopped.compareAndSet(false, true)) {
       LOG.trace("Stopping " + getClass().getSimpleName());
       syncWorkerExecutor.shutdown();
+      txWorkerExecutor.shutdown();
       scheduler.shutdown();
+      servicesExecutor.shutdown();
+      computationExecutor.shutdown();
       shutdown.countDown();
     } else {
       LOG.trace("Attempted to stop already stopped " + getClass().getSimpleName());
     }
   }
 
-  public void awaitStop() throws InterruptedException {
+  void awaitStop() throws InterruptedException {
     shutdown.await();
     if (!syncWorkerExecutor.awaitTermination(2L, TimeUnit.MINUTES)) {
       LOG.error("{} worker executor did not shutdown cleanly.", this.getClass().getSimpleName());
       syncWorkerExecutor.shutdownNow();
       syncWorkerExecutor.awaitTermination(2L, TimeUnit.MINUTES);
     }
+    if (!txWorkerExecutor.awaitTermination(2L, TimeUnit.MINUTES)) {
+      LOG.error(
+          "{} transaction worker executor did not shutdown cleanly.",
+          this.getClass().getSimpleName());
+      txWorkerExecutor.shutdownNow();
+      txWorkerExecutor.awaitTermination(2L, TimeUnit.MINUTES);
+    }
     if (!scheduler.awaitTermination(2L, TimeUnit.MINUTES)) {
       LOG.error("{} scheduler did not shutdown cleanly.", this.getClass().getSimpleName());
       scheduler.shutdownNow();
       scheduler.awaitTermination(2L, TimeUnit.MINUTES);
     }
+    if (!servicesExecutor.awaitTermination(2L, TimeUnit.MINUTES)) {
+      LOG.error("{} services executor did not shutdown cleanly.", this.getClass().getSimpleName());
+      servicesExecutor.shutdownNow();
+      servicesExecutor.awaitTermination(2L, TimeUnit.MINUTES);
+    }
+    if (!computationExecutor.awaitTermination(2L, TimeUnit.MINUTES)) {
+      LOG.error(
+          "{} computation executor did not shutdown cleanly.", this.getClass().getSimpleName());
+      computationExecutor.shutdownNow();
+      computationExecutor.awaitTermination(2L, TimeUnit.MINUTES);
+    }
```

Comment:
This is starting to add up to a lot of waiting.  Maybe we should set it up so that we call `shutdown` on all of them, wait then `shutdownNow` on any not shutdown, then wait and log any that still failed to shutdown?

Reply:
sounds like a good first issue - https://github.com/PegaSysEng/pantheon/issues/768